src/widgets/calendar/lv_calendar.c

```diff
@@ -282,8 +282,10 @@ static void lv_calendar_constructor(const lv_obj_class_t * class_p, lv_obj_t * o
     lv_obj_set_flex_flow(obj, LV_FLEX_FLOW_COLUMN);
     lv_obj_set_flex_grow(calendar->btnm, 1);
 
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_calendar_set_showed_date(obj, calendar->showed_date.year, calendar->showed_date.month);
```

Comment:
Here we just apply what was set in line 246-252. So we should either remove those or keep these lines too.  I vote for removing line 246-252 too.